sdk/servicebus/service-bus/src/core/managementClient.ts

```diff
@@ -390,9 +380,7 @@ export class ManagementClient extends LinkEntity {
       // false without ever having cleared the timeout otherwise.
       clearTimeout(this._tokenRenewalTimer as NodeJS.Timer);
       if (this._isMgmtRequestResponseLinkOpen()) {
-        const mgmtLink = this._mgmtReqResLink;
-        this._mgmtReqResLink = undefined;
-        await mgmtLink!.close();
+        await this.closeLink("permanently");
```

Comment:
`Management.close()` is called in 2 occasions:
- User calls `ServiceBusClient.close()`
- We get a connection disconnected event, and so we call `close()` to clean up timers and close the underlying link

In the first case, we expect any subsequent call to the methods in ManagementClient to throw an error which is handled by `throwErrorIfConnectionClosed()` you see in every method here.

In the second case, any subsequent calls to the methods in ManagementClient will create a new $management link in a new AMQP connection if one is not already created by virtue of any send/receive calls

With the changes in this PR, we now mark the close() call as "permanent". Meaning subsequent calls see the promise returned by `initLink()` get resolved, but no link is created.

So, we should pass `linkOnly` and not `permanently` here

Reply:
Ah, I see - that's actually a little gross. We need an equivalent "onDetached" idea here as well (ie, not closed permanently). 

I'll change it "linkOnly" but it seems like ManagementLink broke the defacto contract. :|

Reply:
Done, also added comment.